### PR TITLE
DEVX-4572 Add Android push notification upload tool docs

### PR DESCRIPTION
**Pull request by igorwojda**
https://jira.vonage.com/browse/DEVX-4572

- add info about Android uploader tool (repo https://github.com/nexmo-community/android-push-uploader)
- tiny updates on iOS side as well

### CODE REVIEW

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 1]
```diff
@@ -131,6 +140,8 @@ curl -v -X PUT \
    https://api.nexmo.com/v1/applications/$VONAGE_APP_ID/push_tokens/android  
 ```
 
+If all the values are correct you should see `200` response code in the terminal.
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:143]: Not entirely correct - there is no validation at this point. 200 means that Vonage got your data and stored it but haven't actually checked that they are working. Worth making that clearer.

File: _documentation/en/client-sdk/setup/set-up-push-notifications/ios.md [hunk 2]
```diff
@@ -54,9 +54,9 @@ You upload your certificate to the Vonage servers by making a POST request. The
 
 To use the tool you will need to run it locally or deploy it. You can follow the the instructions in the GitHub project's [README](https://github.com/nexmo-community/ios-push-uploader#running-the-project). You will also need the private key for your Vonage Application. 
 
-Once you have the tool running, enter your Vonage Application ID, private key, and certificate file and upload. The status of your upload will be shown on the page once it is complete. 
+Once you have the tool running, enter your Vonage Application ID, private key file (located in nexmo application folder), and certificate file and click upload. The status of your upload will be shown on the page once it is complete:
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/ios.md:57]: I don't believe we mentioned any application folder so we should remove the mention of it.

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 3]
```diff
@@ -67,56 +69,63 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
 
 ## Link the Vonage backend push service with the Firebase application
 
-To link the Vonage backend push service with the Firebase application you need to make a single request.
+You link Vonage backend push service with the Firebase application by making a POST request. You can to this request using Upload Tool or making a POST request directly.
+### Using the Upload Tool
+
+The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), does so with a user interface.
+
+Once you have the tool running, enter your Vonage Application ID, private key file (located in nexmo application folder), Firebase project id, Firebase token and click upload. The status of your upload will be shown on the page once it is complete:
```
[the remark this answers is not in the document]
Comment (pardel) [reply _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:107]: Also, I don't believe we mentioned any application folder so we should remove the mention of it.

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 4]
```diff
@@ -67,56 +69,63 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
 
 > **NOTE:** In order to apply any methods on the `NexmoClient` object (for example answer a call, hangup, and so on), the `NexmoClient` has to be initialized and the user has to be [logged in](/client-sdk/getting-started/add-sdk-to-your-app/android) to it.
 
-## Connect Vonage API application to Firebase
+## Link Vonage backend push service to Firebase
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:72]: I believe the original title worked better - we're only affecting change to one application and not the entore Vonage system. I would also not call it a backend - will be confusion for readers.

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 5]
```diff
@@ -67,56 +69,63 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
-```bash
-nexmo jwt:generate ./private.key exp=$(($(date +%s)+86400)) acl='{"paths":{"/*/users/**":{},"/*/conversations/**":{},"/*/sessions/**":{},"/*/devices/**":{},"/*/image/**":{},"/*/media/**":{},"/*/applications/**":{},"/*/push/**":{},"/*/knocking/**":{},"/*/legs/**":{}}}' application_id=VONAGE_APP_ID
-```
-
-> **NOTE** The above commands set the expiry of the JWT to one day from now, which is the maximum.
-
-> **NOTE** A `VONAGE_DEV_JWT` is a JWT without a sub claim. 
-
-> **NOTE:** More details on how to generate a JWT can be found in the [setup guide](/tutorials/client-sdk-generate-test-credentials#generate-a-user-jwt).
+Obtain your Vonage API Application id. You can access the existing application in the [dashboard](https://dashboard.nexmo.com/voice/your-applications). If you don't have an application already you can create the new application via [Nexmo CLI](/client-sdk/setup/create-your-application).
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:82]: existing application => existing application**s**

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 6]
```diff
@@ -67,56 +69,63 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
+Get your Firebase project id from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console -> Project settings -> General`.
 
 ![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)
 
 ![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-id.png)
 
 ### Get Firebase token
 
-Get your `FIREBASE_TOKEN` from the Firebase console. Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key. 
+Get your Firebase token from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key.
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:94]: Are we using the private key as a token? Readers might find confusing. If the answer is yes, consider renaming `FIREBASE_TOKEN` to `FIREBASE_PRIVATE_KEY`

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 7]
```diff
@@ -67,56 +69,63 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
 ### Get Firebase token
 
-Get your `FIREBASE_TOKEN` from the Firebase console. Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key. 
+Get your Firebase token from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key. 
 
 ![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)
 
 ![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-token.png)
 
 ## Link the Vonage backend push service with the Firebase application
```
Comment (pardel) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:100]: Same comment as before - Let's use: "Connect your Vonage application to Firebase"

**Review by pardel — COMMENTED** [review]
I've added some suggestions that will increase the clarity.

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md (hunk 6 above)
Comment (igorwojda) [reply _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:94]: It looks like Firebase nomenclature is calling this a `Server key` so renaming to `FIREBASE_SERVER_KEY` will be more accurate (same as a screenshot)

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 8]
```diff
@@ -67,70 +69,84 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
+![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)
 
-### Get Firebase project Id
+![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-token.png)
 
-Get your `FIREBASE_PROJECT_ID` from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console -> Project settings -> General`.
+## Connect Vonage API application to Firebase
 
-![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)
+You connect Vonage backend push service with the Firebase application by making a POST request. You can to this request using Upload Tool or making a POST request directly.
```
Comment (abdulajet) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:103]:
```suggestion
You connect Vonage backend push service with the Firebase application by making a POST request. You can to this request using the Upload Tool or making a POST request directly.
```

File: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md [hunk 9]
```diff
@@ -67,70 +69,84 @@ source: '_tutorials_tabbed_content/client-sdk/setup/push-notifications/android/f
...
 
-![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-id.png)
+### Using the Upload Tool
 
-### Get Firebase token
+The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), does so with a user interface.
```
Comment (abdulajet) [inline _documentation/en/client-sdk/setup/set-up-push-notifications/android.md:107]:
```suggestion
The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), allows you to upload with a user interface.
```

**Review by abdulajet — APPROVED** [review]

**Review by pardel — APPROVED** [review]

**Pull request merged**